### PR TITLE
refactor: events listing page

### PR DESCRIPTION
**Pull request by Prerna290**
## Type
- ( ) Fix
- (X) Refactor
- ( ) Style
- ( ) Docs
- ( ) Feat
- ( ) Hotfix

### CODE REVIEW

File: NX
Comment (ArshdeepGrover) [inline NX:1]: Please remove this!

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-featured-communities-card/public-home-list-events-featured-communities-card.component.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<div>
+  <a [routerLink]="[]" target="_blank">
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-featured-communities-card/public-home-list-events-featured-communities-card.component.html]: RouterLink was empty?

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-featured-communities-card/public-home-list-events-featured-communities-card.component.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
+a {
+  @apply com-no-underline;
+
+  .featured-community {
+    @apply com-flex com-items-start;
+
+    .avatar {
+      @apply com-h-14 com-w-14;
+      @apply com-rounded-none com-object-cover com-object-center com-align-middle com-mr-4;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-featured-communities-card/public-home-list-events-featured-communities-card.component.scss]: We are using two lines here?

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-past-card/public-home-list-events-past-card.component.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="past-events-container">
+  <a [routerLink]="[]">
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-past-card/public-home-list-events-past-card.component.html]: routerLink was empty?

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

File: NX
Comment (ArshdeepGrover) [inline NX:1]: Let's remove this file

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-past/public-home-list-events-past.component.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  selector: 'commudle-public-home-list-events-past',
+  templateUrl: './public-home-list-events-past.component.html',
+  styleUrls: ['./public-home-list-events-past.component.scss'],
+})
+export class PublicHomeListEventsPastComponent implements OnInit {
+  pastEvents: IEvent[] = [];
+  faCalendarCheck = faCalendarCheck;
+
+  page_info: IPageInfo;
+  total;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-past/public-home-list-events-past.component.ts]: Lets' define data type!

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-speakers/public-home-list-events-speakers.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  selector: 'commudle-public-home-list-events-speakers',
+  templateUrl: './public-home-list-events-speakers.component.html',
+  styleUrls: ['./public-home-list-events-speakers.component.scss'],
+})
+export class PublicHomeListEventsSpeakersComponent implements OnInit {
+  faMicrophone = faMicrophone;
+  speakers: IUser[] = [];
+
+  page_info: IPageInfo;
+  total;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-speakers/public-home-list-events-speakers.component.ts]: Let's define data type

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-upcoming/public-home-list-events-upcoming/public-home-list-events-upcoming.component.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  templateUrl: './public-home-list-events-upcoming.component.html',
+  styleUrls: ['./public-home-list-events-upcoming.component.scss'],
+})
+export class PublicHomeListEventsUpcomingComponent implements OnInit {
+  community: ICommunity;
+  upcomingEvents: IEvent[] = [];
+  faCalendarDays = faCalendarDays;
+
+  page_info: IPageInfo;
+  total;
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/components/public-home-list-events-upcoming/public-home-list-events-upcoming/public-home-list-events-upcoming.component.ts]: Let's define data type

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/services/public-home-list-events.service.ts
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/services/public-home-list-events.service.ts:1]: Are we using this service?

File: apps/shared-modules/mini-user-profile/components/profile-cards/user-profile-card-large/user-profile-card-large.component.html [hunk 7]
```diff
@@ -54,6 +54,6 @@
     </div>
   </a>
   <div *ngIf="showFollowButton" [ngClass]="{ 'align-right': alignFollowToRight }" class="followers">
-    <app-user-follow [username]="user.username" class="follow-btn"></app-user-follow>
+    <app-user-follow [username]="user.username" [isMobileWidthFull]="true" Fullclass="follow-btn"></app-user-follow>
```
Comment (ArshdeepGrover) [inline apps/shared-modules/mini-user-profile/components/profile-cards/user-profile-card-large/user-profile-card-large.component.html]: Why are we using this?
what is Fullclass

File: apps/shared-modules/mini-user-profile/components/user-follow/user-follow.component.scss [hunk 8]
```diff
@@ -9,6 +9,10 @@
     &.fixed-width {
       width: 74px;
     }
+
+    &.full-width {
+      @apply com-bg-red-500 com-w-full;
```
Comment (ArshdeepGrover) [inline apps/shared-modules/mini-user-profile/components/user-follow/user-follow.component.scss]: we are using the background colour red?

File: tailwind.preset.js [hunk 9]
```diff
@@ -34,12 +34,11 @@ module.exports = {
         'Blue-Violet': '#6F22DF',
         Blond: '#FFF1BB',
         'Black-Coral': '#595867',
-        AuroMetalSaurus: '#667085',
         Charcoal: '#344054',
         'Auro-Metal-Saurus': '#667085',
         'Dark-Jungle-Green': '#101828',
         'Light-Silver': '#D0D5DD',
-        'Bright-Gray': '#EAECF0',
+        'Bright-Gray-Light': '#EAECF0',
```
Comment (ArshdeepGrover) [inline tailwind.preset.js:41]: we are renaming this?

File: tailwind.preset.js [hunk 10]
```diff
@@ -34,12 +34,11 @@ module.exports = {
         'Blue-Violet': '#6F22DF',
         Blond: '#FFF1BB',
         'Black-Coral': '#595867',
-        AuroMetalSaurus: '#667085',
```
Comment (ArshdeepGrover) [inline tailwind.preset.js:37]: are we not using this further?

File: apps/shared-models/sessions.model.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+    id: string;
+    name: string;
+    start_time: Date;
+    kommunity: {
+      id: string;
+    };
+  };
+  embedded_content: string;
+  title: string;
+}
```
Comment (ArshdeepGrover) [inline apps/shared-models/sessions.model.ts]: @arpansac is this a good way?
or we are using I Event

File: apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html:1]: Let's rename this component name!

File: apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+          <button nbButton status="info" shape="rectangle" size="medium" outline class="register-button">
+            Register
+          </button>
+        </div>
+      </a>
+    </div>
+  </nb-card-body>
+</nb-card>
+<div class="border"></div>
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html]: Let optimise this in good way with edge case?
1. what if there is only one card is shown?
2. Is this have to show border everywhere

File: apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/services/public-home-list-events.service.spec.ts
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/listing-pages/public-home-list-events/services/public-home-list-events.service.spec.ts:1]: Let's remove this file

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

File: apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html
Comment (Prerna290) [reply apps/commudle-admin/src/app/app-shared-components/events-upcoming-card/events-upcoming-card.component.html:1]: event-horizontal-card

**Pull request merged**